rustls/src/crypto/signer.rs

```diff
@@ -111,4 +118,16 @@ impl CertifiedKey {
             .first()
             .ok_or(Error::NoCertificatesPresented)
     }
+
+    /// Verify the consistency of this [`CertifiedKey`]'s public and private keys.
+    /// This is done by performing a comparison of SubjectPublicKeyInfo bytes.
+    pub fn certificate_and_key_are_consistent(&self) -> Result<bool, Error> {
```

Comment:
Nit: this should be moved before `end_entity_cert()`.

This name also seems pretty verbose? Maybe `keys_match()` or `matching_keys()`?

Reply:
Moved and renamed to `keys_match()` in [b3577ab](https://github.com/rustls/rustls/pull/1954/commits/b3577abc6b66c77c0e5024adcd8248f8f70115f7)!